R/ttestindependentsamples.R

```diff
@@ -355,9 +363,14 @@ ttestIndependentMainTableRow <- function(variable, dataset, test, testStat, effS
   ## assumption is met; seems like in this setting there is no
   ## sampling plan, thus the p-value is not defined. haha!
   leveneViolated <- FALSE
-  if (!optionsList$wantsWelchs && !optionsList$wantsWilcox && optionsList$wantsStudents) {
+  if (!optionsList$wantsWelchs && !optionsList$wantsWilcox && optionsList$wantsStudents && options$equalityOfVarianceType == "brownForsythe") {
+    levene <- car::leveneTest(variableData, groupingData, "median")
+    ## arbitrary cut-offs are arbitrary
+    if (!is.na(levene[1, 3]) && levene[1, 3] < 0.05)
+      leveneViolated <- TRUE
+    }
+  if (!optionsList$wantsWelchs && !optionsList$wantsWilcox && optionsList$wantsStudents && options$equalityOfVarianceType == "levene") {
```

Comment:
This part seems to duplicate the code unnecesarily, as you only need to switch between median and mean, e.g.,

```r
center <- if(options$equalityOfVarianceType == "brownForsythe") "median" else "mean"
levene <- car::leveneTest(variableData, groupingData, center)
```
instead of duplicatin the `if` `else` blocks.